src/control/tools/BaseStrokeHandler.cpp

```diff
@@ -15,58 +15,11 @@ guint32 BaseStrokeHandler::lastStrokeTime;  // persist for next stroke
 
 BaseStrokeHandler::BaseStrokeHandler(XournalView* xournal, XojPageView* redrawable, const PageRef& page, bool flipShift,
                                      bool flipControl):
-        InputHandler(xournal, redrawable, page) {
+        InputHandler(xournal, redrawable, page), SnapToGridInputHandler(xournal->getControl()->getSettings()) {
```

Comment:
Use composition over inheritance here, since this is a "has-an" not an "is-an" relationship.